api-presigned.go

```diff
@@ -84,25 +84,40 @@ func (c Client) presignURL(method string, bucketName string, objectName string,
 }
 
 // PresignedGetObject - Returns a presigned URL to access an object
-// without credentials. Expires maximum is 7days - ie. 604800 and
-// minimum is 1. Additionally you can override a set of response
-// headers using the query parameters.
+// data without credentials. URL can have a maximum expiry of
+// upto 7days or a minimum of 1sec, if expiry duration is set to 0
+// URL a maximum value of 7 days is automatically choosen. Additionally
+// you can override a set of response headers using the query parameters.
 func (c Client) PresignedGetObject(bucketName string, objectName string, expires time.Duration, reqParams url.Values) (u *url.URL, err error) {
 	return c.presignURL("GET", bucketName, objectName, expires, reqParams)
 }
 
-// PresignedHeadObject - Returns a presigned URL to access an object metadata
-// without credentials. Expires maximum is 7days - ie. 604800 and minimum is 1.
+// PresignedHeadObject - Returns a presigned URL to access object
+// metadata without credentials. URL can have a maximum expiry of
+// upto 7days or a minimum of 1sec, if expiry duration is set to 0
+// URL a maximum value of 7 days is automatically choosen.
+// Additionally you can override a set of response headers using
+// the query parameters.
 func (c Client) PresignedHeadObject(bucketName string, objectName string, expires time.Duration, reqParams url.Values) (u *url.URL, err error) {
 	return c.presignURL("HEAD", bucketName, objectName, expires, reqParams)
 }
 
-// PresignedPutObject - Returns a presigned URL to upload an object without credentials.
-// Expires maximum is 7days - ie. 604800 and minimum is 1.
+// PresignedPutObject - Returns a presigned URL to upload an object
+// without credentials. URL can have a maximum expiry of upto 7days
+// or a minimum of 1sec, if expiry duration is set to 0 URL a
```

Comment:
I didn't test it but after reading the code, it seems setting 0 to expires returns an error (invoked by isValidExpiry() in presignURL())

Reply:
I see looks like behavior has changed.. @vadmeste - let me change the comment to reflect the new changes.